src/context.cc

```diff
@@ -144,24 +144,28 @@ ValuePtr Context::lookup_variable(const std::string &name, bool silent) const
 		return this->variables.find(name)->second;
 	}
 	if (this->parent) {
-		return this->parent->lookup_variable(name, silent);
+		return this->parent->lookup_variable(name, silent, loc);
 	}
 	if (!silent) {
-		PRINTB("WARNING: Ignoring unknown variable '%s'.", name);
+		if(loc.isNone()){
+			PRINTB("WARNING: Ignoring unknown variable '%s'.", name);
+		}else{
+			PRINTB("WARNING: Ignoring unknown variable '%s', line %d.", name% loc.firstLine());
```

Comment:
Is there a nice way we could avoid this repetition? I fear this may result in the two strings getting out of sync at some point.

Reply:
Work in progress. I have a good idea.

In theory, everything that can cause an error should have a location associated.
I still want to specifically handle the "location none" case as a fail safe, particularly when I later want to access the path/file (which is a nullpointer for location none), but the warning/errors without a line number should be rarely seen.

On a side note: I assume that translating errors and warnings is not a mid-therm goal - concatenating strings is easy, but can create awkward sentences/construct to translate.